src/library/scala/collection/immutable/Range.scala

```diff
@@ -155,15 +155,16 @@ extends scala.collection.AbstractSeq[Int]
   @inline final override def foreach[@specialized(Unit) U](f: Int => U) {
     validateMaxLength()
     val isCommonCase = (start != Int.MinValue || end != Int.MinValue)
-    var i = start
+    var i: Long = start
     var count = 0
-    val terminal = terminalElement
-    val step = this.step
+    val step: Long = this.step.toLong
+    val terminal: Long = lastElement.toLong + step
```

Comment:
Is there any way to fix this issue without `Long`s? This implementation will literally kill the performance of `foreach` on Scala.js, because longs are excruciatingly slow.

What would be the consequences of turning the loop into an
```scala
if (!isEmpty) {
  do {
    ...
  } while (
    ...
  )
}
```
? Would that be bad for bounds check removal or some other reason?